spec/controllers/shared_accesses_controller_spec.rb

```diff
@@ -55,27 +57,47 @@
   end
 
   describe '#shareable_users' do
-    it 'returns the users that the room can be shared to' do
+    it 'returns no users since they dont have SharedList permission' do
       room = create(:room)
-      shared_users = create_list(:user, 5)
+      room.shared_users = create_list(:user, 5)
       shareable_users = create_list(:user, 5)
       shareable_users << user
 
-      room.shared_users = shared_users
-
       get :shareable_users, params: { friendly_id: room.friendly_id, search: '' }
       response_users_ids = JSON.parse(response.body)['data'].map { |user| user['id'] }
-      expect(response_users_ids).to match_array(shareable_users.pluck(:id))
+      expect(response_users_ids).to match_array([])
     end
 
-    it 'returns the shareable users according to the query' do
-      room = create(:room)
-      room.shared_users = create_list(:user, 5)
-      shareable_users = create_list(:user, 5, name: 'Jane Doe')
+    context 'users are given SharedList permission' do
+      before do
+        user.update!(role: shared_list_role)
+        create(:role_permission,
+               role: shared_list_role,
+               permission: shared_list_permission,
+               value: 'true',
+               provider: 'greenlight')
```

Comment:
Could we put this on top of the file like the others specs please

Reply:
It's a bit different since it isn't being assigned to a variable in this file. That's why I didn't put it at the top like the rest of the files, rubocop complains. I could put it in the `before do` at the very top but I felt it would be better to put it in this more specific `before do`